TS_TSX.md

```diff
@@ -0,0 +1,48 @@
+# File Metrics for TypeScript (TS) and TypeScript JSX (TSX)
+
+This document outlines the definition of file metrics for TypeScript (TS) and TypeScript JSX (TSX).
+
+### 1. classes
+
+The "classes" metric counts:
+
+-   keywords like: `class`, `abstract class`, `interface`, `enum`
+-   type aliases with object type literals e.g. `type ObjectType = { message: string};`
+
+It does **not** count:
+
+-   other type aliases like
+    -   `type Union = {a: 1} & {b: 2};`
+    -   `type Intersect = {a: 1} | {b: 2};`
+    -   `type Tuple = [name: string, age: number];`
+    -   `type AliasType = number;`
+
+### 2. functions
+
+The "functions" metric counts:
+
+-   methods declarations in classes, abstract classes and interfaces
+-   static methods, constructors, getters and setters
+-   functions declared with the keyword `function`
+-   arrow functions
+
+### 3. complexity
+
+The "complexity" metric counts:
+
+-   `if`, `if else` statements, but **no** `else` statements
```

Comment:
I don't know if we should explicitely mention that we count no `else` here, as that is already clear when reading the metric definition in the readme and when mentioning that we count for `if` and `if else`.

Reply:
The same holds for `default` and `finally` below